### ISSUE TITLE
Accessibility: ng-select ARIA roles and controls (combobox, listbox, option) improperly set

### ISSUE THREAD

**Issue by isaacfimbres1** [ng-select/ng-select#1617]
**Describe the bug**
The following aria roles on the searchable multi-select dropdown are improperly set: combobox, listbox, and option.  Aria-controls is also not set on the text input element.

**To Reproduce** 
Steps to reproduce the behavior (if example is not provided):
1. Render the ng-select component, as long as some options are populated, there's nothing special that needs to happen. The following properties were used to configure the component
```
<ng-select [items]="options"
      [bindLabel]="labelProp"
      [multiple]=true
      placeholder={{placeholder}}
      [formControl] ="formControl"
      labelForId={{id}}
      aria-label="multiselect-dropdown-widget"
      class="custom"
      >
```

2. The accessibility assessment was done using the FastPass feature which comes with the 'Microsoft Accessibility Insights for Web' extension, available on chrome and edge.
3. Once the fastpass is run, the `aria-required-children` check will fail. There are two cases where this fails, which demonstrates the incorrect configuration of aria roles. See screenshots below.
 
**Expected behavior**
Based on WCAG 1.3.1 standards, the input text box should be contained in a div with `role="combobox"`, and the options should be contained in a div with `role="listbox"`. Furthermore, the input text box should have an attribute `aria-controls` which is linked to the listbox id, similar to how `aria-owns` property is already used. Currently, the entire ng-select component has the `role="listbox"` while the input element itself has `role="combobox"`.

This links gives an overview of the rule in question
https://dequeuniversity.com/rules/axe/3.5/aria-required-children?application=msftAI

This links provides a good example of the correct way to set these roles
https://www.w3.org/TR/wai-aria-1.1/#combobox

**Screenshots**
This is the check which fails when the dropdown is not open:
![image](https://user-images.githubusercontent.com/23004170/81850537-fb827b00-950c-11ea-8999-e4c596372658.png)

This is the check which fails when the dropdown is opened:
![image](https://user-images.githubusercontent.com/23004170/81850638-1f45c100-950d-11ea-926a-94d91943c1f6.png)

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser: Edge (chromium)
 - Version: 3.5.0

**Comment by geebru** [comment]
My PR addresses these #1498 - I haven't had time to continue work on it and won't for the forseeable future but feel free to help move these a11y improvements along, or help get the PR working from a testing vantage.

**Comment by isaacfimbres1** [comment]
I looked at @geebru's PR but unfortunately did not have the bandwidth to build off of it as there are a good amount of changes. I've made this incremental update to fix the aria markup for the time being
https://github.com/ng-select/ng-select/pull/1673

**Comment by AElmoznino** [comment]
Any updates on this? I would love to have 100 a11y score when using ng-select 😄

**Comment by Hoopou** [comment]
Still not fixed!

**Comment by isaacfimbres1** [comment]
I already have an approval on this PR https://github.com/ng-select/ng-select/pull/1673, just need someone with write access to approve!
